code/modules/power/singularity/collector.dm

```diff
@@ -163,3 +163,90 @@ var/global/list/rad_collectors = list()
 		flick("ca_deactive", src)
 	update_icons()
 	return
+
+/obj/machinery/power/rad_collector/pipenet
+	name = "Modular Radiation Collector Array"
+	desc = "A variant of the radiation collector array which connects to a pipecap underneath it."
+	var/obj/machinery/atmospherics/pipe/cap/pipe = null
+	var/datum/gas_mixture/removed = null
+
+/obj/machinery/power/rad_collector/pipenet/Destroy()
+	QDEL_NULL(removed)
+	pipe = null
+	. = ..()
+
+/obj/machinery/power/rad_collector/pipenet/Initialize()
+	if(anchored)
+		locate_pipenet()
+		connect_to_network()
+
```

Comment:
It's that time again. All Initialize()'s must return their parent's init hint or a custom one.